snooty/rstparser.py

```diff
@@ -1072,29 +1072,23 @@ def register_spec_with_docutils(
         )
         builder.add_directive(name, DocutilsDirective)
 
+    # copy tabs directive declaration
+    tabs_directive = spec.directive["tabs"]
+    tabs_directive.options["tabid"] = specparser.PrimitiveType.string
+
     # Define tabsets
     for name in spec.tabs:
-        tabs_name = "tabs-" + name
         tabs_base_class: Any = BaseTabsDirective
-        directive = specparser.Directive(
-            inherit=None,
-            help=None,
-            example=None,
-            content_type="block",
-            argument_type=None,
-            required_context=None,
-            domain=None,
-            name=tabs_name,
-            options={"tabid": specparser.PrimitiveType.string},
-        )
+        tabs_name = "tabs-" + name
+        tabs_directive.name = tabs_name
```

Comment:
Here's where you would actually want to copy it. The way you would do this might be:
```
modified_tabs_directive = dataclasses.replace(tabs_directive, name=tabs_name)
```

using [dataclasses.replace](https://docs.python.org/3/library/dataclasses.html#dataclasses.replace)